spinnaker/spinnaker_system/google_http_lb_upsert_test.py

```diff
@@ -0,0 +1,894 @@
+      'requestPath': '/',
+      'port': 80,
+      'checkIntervalSec': 1,
+      'timeoutSec': 1,
+      'healthyThreshold': 1,
+      'unhealthyThreshold': 1
+    }
+    self.__proto_upsert = {
+      'cloudProvider': 'gce',
+      'provider': 'gce',
+      'stack': bindings['TEST_STACK'],
+      'credentials': bindings['GCE_CREDENTIALS'],
+      'region': bindings['TEST_GCE_REGION'],
+      'loadBalancerType': 'HTTP',
+      'loadBalancerName': self.__lb_name,
+      'portRange': '80',
+      'defaultService': {
+        'name': 'default-backend-service',
+        'backends': [],
+        'healthCheck': self.__proto_hc,
+      },
+      'certificate': '',
+      'hostRules': [
+        {
+          'hostPatterns': ['host1.com', 'host2.com'],
+          'pathMatcher': {
+            'pathRules': [
+              {
+                'paths': ['/path', '/path2/more'],
+                'backendService': {
+                  'name': 'backend-service',
+                  'backends': [],
+                  'healthCheck': self.__proto_hc,
+                }
+              }
+            ],
+            'defaultService': {
+              'name': 'pm-backend-service',
+              'backends': [],
+              'healthCheck': self.__proto_hc,
+            }
+          }
+        }
+      ],
+      'type': 'upsertLoadBalancer',
+      'availabilityZones': {bindings['TEST_GCE_REGION']: []},
+      'user': '[anonymous]'
+    }
+
+
+  def _make_bs_link(self, bs):
+    '''Make a fully-formatted backend service link.
+    '''
+    return (GCE_URL_PREFIX + self.bindings['GCE_PROJECT'] +
+            '/global/backendServices/' + bs)
+
+
+  def _make_hc_link(self, hc):
+    '''Make a fully-formatted health check link.
+    '''
+    return (GCE_URL_PREFIX + self.bindings['GCE_PROJECT'] +
+            '/global/httpHealthChecks/' + hc)
+
+
+  def _set_all_hcs(self, upsert, hc):
+    '''Set all health checks in upsert to hc.
+    '''
+    upsert['defaultService']['healthCheck'] = hc
+    for host_rule in upsert['hostRules']:
+      path_matcher = host_rule['pathMatcher']
+      path_matcher['defaultService']['healthCheck'] = hc
+      for path_rule in path_matcher['pathRules']:
+        path_rule['backendService']['healthCheck'] = hc
+
+
+  def _make_contract_clauses(self, contract_builder, upsert):
+    '''Add the proper predicates to the contract builder for a given
+    upsert description.
+    '''
+    host_rules = upsert['hostRules'] # Host rules will be distinct.
+    backend_services = [upsert['defaultService']]
+    for host_rule in host_rules:
+      path_matcher = host_rule['pathMatcher']
+      backend_services.append(path_matcher['defaultService'])
+      for path_rule in path_matcher['pathRules']:
+        backend_services.append(path_rule['backendService'])
+    health_checks = [service['healthCheck'] for service in backend_services]
+
+    hc_clause_builder = contract_builder.new_clause_builder('Health Checks Created',
```

Comment:
I'd combine these two to
hc_clause_bulider = contract_builder.new()
   .list_resource('httpHealthChecks')